src/main/cpp/commands/MoveToLevelCommand.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "commands/MoveToLevelCommand.h"
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.
+// For more information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+
+MoveToLevelCommand::MoveToLevelCommand(t34::Elevator *elevator, t34::AlgaeIntake *algae_intake, t34::CoralIntake *coral_intake, int level) 
+: m_elevator(elevator)
+, m_algae_intake(algae_intake)
+, m_coral_intake(coral_intake)
+, m_level(level)
+
+{
+
+  m_level = std::clamp(level, 0, static_cast<int>(presets.size()) - 1);
+    
+  auto [algae_angle, coral_angle, elevator_height] = presets.at(level);
```

Comment:
Bug: You're not passing the clamped value to `presets.at()`.

```suggestion
  auto [algae_angle, coral_angle, elevator_height] = presets.at(m_level);
```